core/eolearn/tests/test_eodata_io.py

```diff
@@ -121,26 +123,47 @@ def test_overwriting_non_empty_folder(eopatch, fs_loader):
 
 @mock_s3
 @pytest.mark.parametrize("fs_loader", FS_LOADERS)
-def test_save_load(eopatch, fs_loader):
+@pytest.mark.parametrize(
+    "save_kwargs, load_kwargs",
+    [
+        ({"path": "/random_path"}, {"path": "/random_path"}),
+        (
+            {"path": "/", "overwrite_permission": OverwritePermission.OVERWRITE_FEATURES},
+            {"path": "/", "lazy_loading": False},
+        ),
+    ],
+)
+def test_save_load(eopatch: EOPatch, fs_loader, save_kwargs: Dict[str, Any], load_kwargs: Dict[str, Any]):
     with fs_loader() as temp_fs:
-        eopatch.save("/", filesystem=temp_fs)
-        eopatch2 = EOPatch.load("/", filesystem=temp_fs)
-        assert eopatch == eopatch2
-
-        eopatch2.save("/", filesystem=temp_fs, overwrite_permission=1)
-        eopatch2 = EOPatch.load("/", filesystem=temp_fs)
-        assert eopatch == eopatch2
-
-        eopatch2.save("/", filesystem=temp_fs, overwrite_permission=1)
-        eopatch2 = EOPatch.load("/", filesystem=temp_fs, lazy_loading=False)
-        assert eopatch == eopatch2
-
-        features = {FeatureType.DATA_TIMELESS: ["mask"], FeatureType.TIMESTAMPS: ...}
-        eopatch2.save("/", filesystem=temp_fs, features=features, compress_level=3, overwrite_permission=1)
-        eopatch2 = EOPatch.load("/", filesystem=temp_fs, lazy_loading=True)
-        assert eopatch == eopatch2
-        eopatch3 = EOPatch.load("/", filesystem=temp_fs, lazy_loading=True, features=features)
-        assert eopatch != eopatch3
+        eopatch.save(**save_kwargs, filesystem=temp_fs)
+        loaded_eopatch = EOPatch.load(**load_kwargs, filesystem=temp_fs)
+        assert eopatch == loaded_eopatch
+
+
+@mock_s3
+@pytest.mark.parametrize("fs_loader", FS_LOADERS)
+@pytest.mark.parametrize(
+    "save_features, load_features",
+    [
+        (..., ...),
+        (..., ...),
```

Comment:
Is this accidentally duplicated? With the `with` statement below I would imagine the state is reset for each entry in the parametrization list.

If the state is not reset, I would expect this to fail, since the default overwrite_permission is ADD_ONLY.

Reply:
it's an oopsie :see_no_evil: